src/main/java/org/spongepowered/common/data/processor/data/entity/TagsDataProcessor.java

```diff
@@ -63,8 +62,7 @@ protected boolean set(Entity dataHolder, Set<String> value) {
         Set<String> added = new HashSet<String>();
```

Comment:
Do you have a benchmark for that? Linked data structures incur a lot of cache misses and normally perform worst on modern CPU than arrays, even in insertion and deletion.